src/auth/storacha.ts

```diff
@@ -0,0 +1,151 @@
+import * as Client from "@web3-storage/w3up-client";
+import { Config, readConfig, writeConfig } from "../utils/config";
+import inquirer from "inquirer";
+import ora from "ora";
+import chalk from "chalk";
+import { OwnedSpace, Space } from "@web3-storage/w3up-client/dist/src/space";
+
+export class StorachaAuth {
+  private config: Config;
+
+  constructor() {
+    this.config = readConfig();
+  }
+
+  async login(): Promise<Client.Client> {
+    const { email } = await inquirer.prompt([
+      {
+        type: "input",
+        name: "email",
+        message: "Enter your email:",
+        validate: (input) => {
+          if (!input.includes("@")) {
+            return "Please enter a valid email address";
+          }
+          return true;
+        },
+      },
+    ]);
+
+    const spinner = ora("Logging in to Storacha...").start();
+    spinner.color = "red";
+
+    try {
+      const client = await Client.create();
+      await client.login(email);
+
+      writeConfig({
+        ...this.config,
+        storacha: {
+          ...this.config.storacha,
+          email,
+        },
+      });
+      spinner.succeed(`Successfully logged in to Storacha with ${email}`);
+      return client;
+    } catch (error) {
+      spinner.fail(
+        `\nFailed to login to Storacha: ${(error as Error).message}`,
+      );
+      console.log(chalk.yellow("  Troubleshooting:"));
+      console.log(chalk.yellow("  1.") + " Check your internet connection");
+      console.log(
+        chalk.yellow("  2.") + " Verify your email is registered with Storacha",
+      );
+      console.log(
+        chalk.yellow("  3.") +
+          " You may need to check your email for a verification link",
+      );
+      console.log(
+        chalk.yellow("  4.") +
+          " Ensure that you've also connected your card for billing. The free plan, at last",
+      );
+      throw error;
+    }
+  }
+
+  async createStorachaSpace(
+    client: Client.Client,
+  ): Promise<OwnedSpace | undefined> {
+    const { spaceName } = await inquirer.prompt([
+      {
+        type: "input",
+        name: "spaceName",
+        message: "Enter a name for your Storacha space",
+        validate: (input) => input.length > 0 || "Space name cannot be blank",
+      },
+    ]);
+
+    const spinner = ora(`Creating "${spaceName}"...`).start();
+    spinner.color = "red";
+
+    try {
+      const space = await client.createSpace(spaceName);
```

Comment:
after creating a space you'll need to set up "recovery" - either giving the user the private key in a readable format or backing up delegations with our service - see https://github.com/storacha/w3cli/blob/3a737891c4ad988e091eca9d0fc9bc96a4f83a6f/space.js#L58 for how that works in the current CLI

Reply:
awseome! i saw this in action when I tried creating a space with w3 CLI. 

I'll check it out.

Reply:
So, @travis, the issue was pretty much on my end. The `createSpace` method from `w3up-client` already handles delegations and the recovery setup internally. 

See it here: 

https://github.com/storacha/w3up/blob/2ddb4042fee59e6b667057f6e82c711fa9e318ee/packages/w3up-client/src/client.js#L293-L300

I forgot to pass the `account` option when I tried calling `createSpace`. When I did now, I could back up to the new space I created. 

I pushed an update just now.